UnicornRender/video/source/CameraFpsController.cpp

```diff
@@ -5,87 +5,59 @@
 */
 
 #include <unicorn/video/CameraFpsController.hpp>
-#include <algorithm>
 
 namespace unicorn
 {
 namespace video
 {
-CameraFpsController::CameraFpsController(unicorn::video::Camera& camera)
-    : sensitivity(0.1f), 
-    speed(100.0f), 
-    m_lastX(0.0), 
-    m_lastY(0.0), 
-    m_yaw(90.0),
-    m_pitch(0.0), 
-    m_dirty(false),
-    m_camera(camera)
-{
-}
 
-void CameraFpsController::MoveUp(float deltaTime) const
+CameraFpsController::CameraFpsController(glm::mat4& cameraView)
+    : CameraController(cameraView)
+    , m_mousePosition(0)
+    , m_dirtyViewPosition(true)
 {
-    m_camera.Translate({0.0f, -deltaTime * speed, 0.0f});
 }
 
-void CameraFpsController::MoveDown(float deltaTime) const
+void CameraFpsController::UpdateView(float x, float y)
 {
-    m_camera.Translate({0.0f, speed * deltaTime, 0.0});
-}
+    const float xoffset = m_mousePosition.x - x;
+    const float yoffset = y - m_mousePosition.y;
 
-void CameraFpsController::MoveLeft(float deltaTime) const
-{
-    m_camera.Translate(-glm::normalize(glm::cross(m_camera.GetDirection(), m_camera.GetUpVector())) * speed * deltaTime);
-}
+    m_mousePosition.x = x;
+    m_mousePosition.y = y;
 
-void CameraFpsController::MoveRight(float deltaTime) const
-{
-    m_camera.Translate(glm::normalize(glm::cross(m_camera.GetDirection(), m_camera.GetUpVector())) * speed * deltaTime);
+    if (m_dirtyViewPosition)
+    {
+        m_dirtyViewPosition = false;
+        return;
+    }
+
+    m_rotation.x += glm::radians(yoffset);
+    m_rotation.y += glm::radians(xoffset);
+
+    m_isDirty = true;
 }
 
-void CameraFpsController::MoveForward(float deltaTime) const
+void CameraFpsController::SetViewPositions(double x, double y)
 {
-    m_camera.Translate(m_camera.GetDirection() * speed * deltaTime);
+    m_mousePosition = glm::vec2(x, y);
 }
 
-void CameraFpsController::MoveBackward(float deltaTime) const
+void CameraFpsController::ResetView()
 {
-    m_camera.Translate(-m_camera.GetDirection() * speed * deltaTime);
+    m_dirtyViewPosition = false;
 }
 
-void CameraFpsController::UpdateView(double posX, double posY)
+void CameraFpsController::CalculateOrientation()
```

Comment:
if it doesn't return the calculated value (and I'm not saying it should), you could name it `UpdateOrientation` instead since that's what it does - it updates orientation and rotation